modules/apps/27-interchain-accounts/types/account.go

```diff
@@ -66,20 +65,24 @@ func NewInterchainAccount(ba *authtypes.BaseAccount, accountOwner string) *Inter
 }
 
 // SetPubKey - Implements AccountI
-func (InterchainAccount) SetPubKey(pubKey crypto.PubKey) error {
+func (ia InterchainAccount) SetPubKey(pubKey crypto.PubKey) error {
 	return fmt.Errorf("not supported for interchain accounts")
 }
 
 // SetSequence - Implements AccountI
-func (InterchainAccount) SetSequence(seq uint64) error {
+func (ia InterchainAccount) SetSequence(seq uint64) error {
 	return fmt.Errorf("not supported for interchain accounts")
 }
 
 func (ia InterchainAccount) Validate() error {
+	if strings.TrimSpace(ia.AccountOwner) == "" {
+		return fmt.Errorf("AccountOwner cannot be empty")
```

Comment:
I think this needs to be an sdk registered error otherwise the error message will be lost when it is converted to an sdk internal error

Reply:
We can only use `fmt.Errorf` for genesis validation or client side code

Reply:
Good catch. Updated this.